services/bundle_analysis/new_notify/contexts/comment.py

```diff
@@ -0,0 +1,171 @@
+        Raises:
+            NotificationContextBuildError: missing some information necessary to create
+                the BundleAnalysisComparison.
+        """
+        if self.is_field_loaded("bundle_analysis_comparison"):
+            return self
+        pull = self._notification_context.pull
+        try:
+            comparison = ComparisonLoader(pull).get_comparison()
+            self._notification_context.bundle_analysis_comparison = comparison
+            return self
+        except (
+            MissingBaseCommit,
+            MissingHeadCommit,
+            MissingBaseReport,
+            MissingHeadReport,
+        ) as exp:
+            raise NotificationContextBuildError(
+                "load_bundle_comparison", detail=exp.__class__.__name__
+            )
+
+    def evaluate_has_enough_changes(self) -> "BundleAnalysisCommentContextBuilder":
+        """Evaluates if the NotificationContext includes enough changes to send the notification.
+        Configuration is done via UserYaml.
+        If a comment was previously made for this PR the required changes are bypassed so that we
+        update the existing comment with the latest information.
+        Raises:
+            NotificationContextBuildError: required changes are not met.
+        """
+        current_yaml = self._notification_context.current_yaml
+        required_changes: bool | Literal["bundle_increase"] = (
+            current_yaml.read_yaml_field(
+                "comment", "require_bundle_changes", _else=False
+            )
+        )
+        changes_threshold: int = current_yaml.read_yaml_field(
+            "comment", "bundle_change_threshold", _else=0
+        )
+        pull = self._notification_context.pull
+        if pull.database_pull.bundle_analysis_commentid:
+            log.info(
+                "Skipping required_changes verification because comment already exists",
+                extra=dict(
+                    pullid=pull.database_pull.id,
+                    commitid=self._notification_context.commit.commitid,
+                ),
+            )
+            return self
+        comparison = self._notification_context.bundle_analysis_comparison
+        should_continue = {
+            False: True,
+            True: abs(comparison.total_size_delta) > changes_threshold,
+            "bundle_increase": (
+                comparison.total_size_delta > 0
+                and comparison.total_size_delta > changes_threshold
+            ),
+        }.get(required_changes, True)
+        if not should_continue:
+            raise NotificationContextBuildError("evaluate_has_enough_changes")
+        return self
+
+    def build_context(self) -> "BundleAnalysisCommentContextBuilder":
+        super().build_context()
+        async_to_sync(self.load_enriched_pull)()
+        self.load_bundle_comparison()
+        self.evaluate_has_enough_changes()
+        return self
```

Comment:
> so you don't really need the builder pattern exactly

You're probably right... I do want to avoid the load function being declared in the Context though... I would prefer to have the context be more similar to a dataclass.

I think using composition for the load steps (they are declared in different classes that are composed into the context) and the context itself declared the build steps might reduce the amount of code we have and overall be more flexible and easier to reuse steps